### PR TITLE
systemd: remove ceph-create-keys from presets

### PR DESCRIPTION
**Pull request by leseb**
ceph-create-keys unit file was removed here:

* https://github.com/ceph/ceph/commit/8bcb4646b6b9846bb965cdec3ca2a21eb3b26bab
* https://github.com/ceph/ceph/commit/dc5fe8d415858358bd0baf5d8dce0a753f5e0cea

As a consequence the systemctl preset command now fails to run since the
unit does not exist anymore. Due to the redirection in /dev/null we
don't know what's happening.

Ultimately the mon unit doesn't get enabled and the mon service won't
start after reboot.
Removing the old/non-existent unit makes the command succeed now.

Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

**Comment by ktdreyer** [comment]
This is happening in an `%if 0%{?suse_version}` conditional. Did you test it on SUSE?

File: ceph.spec.in [hunk 1]
```diff
@@ -1190,7 +1190,7 @@ fi
 %post mon
 %if 0%{?suse_version}
 if [ $1 -eq 1 ] ; then
-  /usr/bin/systemctl preset ceph-create-keys@\*.service ceph-mon@\*.service ceph-mon.target >/dev/null 2>&1 || :
+  /usr/bin/systemctl preset ceph-mon@\*.service ceph-mon.target >/dev/null 2>&1 || :
 fi
 %endif
 %if 0%{?fedora} || 0%{?rhel}
```
Comment (ktdreyer) [inline ceph.spec.in]: does the same thing need to happen in the fedora/rhel case here?

**Comment by leseb** [comment]
I tested this on CentOS.

File: ceph.spec.in (hunk 1 above)
Comment (leseb) [inline ceph.spec.in]: Yes!

**Comment by smithfarm** [comment]
@leseb There are several more mentions of the `ceph-create-keys` unit file in the ceph-mon scriptlets. It would be great if we could remove them all in this one PR.

**Comment by leseb** [comment]
@smithfarm sure, done.

**Comment by vasukulkarni** [comment]
can a build be generated with this change in ceph-ci? i would like to scheduled systemd test from smoke suite.

**Comment by smithfarm** [comment]
@vasukulkarni Pushed `wip-preset-mon` to ceph-ci just now - please let me know if there is any problem.

**Comment by vasukulkarni** [comment]
This test is passing on centOS with one of my other tracker issue fixed as well, but on xenial after reboot i dont see the mon daemon's up 

http://pulpito.ceph.com/vasu-2017-03-30_21:08:42-smoke-wip-preset-mon---basic-vps/

mon's dont come up: https://paste.fedoraproject.org/paste/xf6cWnkRHzkL9U1K1IRLrF5M1UNdIGYhyRLivL9gydE=/raw

**Comment by vasukulkarni** [comment]
If you want to rerun: teuthology-suite -s smoke -c wip-preset-mon -m vps --suite-repo git://github.com/ceph/ceph.git --suite-branch wip-systemd --suite-relpath qa --filter systemd

**Comment by smithfarm** [comment]
The Xenial failure can't be related to this PR, which is limited to `ceph.spec.in`.

**Comment by vasukulkarni** [comment]
thanks, will file an issue for xenial.

**Comment by ktdreyer** [comment]
I don't think we have a tracker to track the jewel backport here?

**Comment by smithfarm** [comment]
@ktdreyer The tracker is http://tracker.ceph.com/issues/19431 but we've still got `systemd/ceph-create-keys@.service` in jewel so I only backported it to kraken for now.

**Comment by ktdreyer** [comment]
Oh, right, thanks!

**Pull request merged**